File: server/modules/git/commit.ts
import { GitError, runGit } from "./exec.js";
import { parseGitStatusPorcelain } from "./status.js";

export function hasStagedChanges(worktreePath: string): boolean {
  const raw = runGit(worktreePath, ["status", "--porcelain"], { trim: false });
  const entries = parseGitStatusPorcelain(raw);
  return entries.some((entry) => entry.staged != null);
}

export function commitStagedChanges(worktreePath: string, message: string): void {
  const trimmed = message.trim();
  if (!trimmed) {
    throw new GitError("Commit message is required", "");
  }
  if (!hasStagedChanges(worktreePath)) {
    throw new GitError("Nothing staged to commit", "");
  }
  runGit(worktreePath, ["commit", "-m", trimmed]);
}

File: server-rs/src/git/commit.rs
use super::exec::{run_git, GitError, RunGitOptions};
use super::status::parse_git_status_porcelain;

pub fn has_staged_changes(worktree_path: &str) -> Result<bool, GitError> {
    let raw = run_git(
        worktree_path,
        &["status", "--porcelain"],
        Some(RunGitOptions { trim: false }),
    )?;
    let entries = parse_git_status_porcelain(&raw);
    Ok(entries.iter().any(|entry| entry.staged.is_some()))
}

pub fn commit_staged_changes(worktree_path: &str, message: &str) -> Result<(), GitError> {
    let trimmed = message.trim();
    if trimmed.is_empty() {
        return Err(GitError::new("Commit message is required", ""));
    }
    if !has_staged_changes(worktree_path)? {
        return Err(GitError::new("Nothing staged to commit", ""));
    }
    run_git(worktree_path, &["commit", "-m", trimmed], None)?;
    Ok(())
}
